packages/babel-preset-react-app/create.js

```diff
@@ -97,11 +117,15 @@ module.exports = function(api, opts, env) {
       ],
     ].filter(Boolean),
     plugins: [
-      // Strip flow types before any other transform, emulating the behavior
+      // Strip flow and typescript types before any other transform, emulating the behavior
       // order as-if the browser supported all of the succeeding features
       // https://github.com/facebook/create-react-app/pull/5182
       isFlowEnabled &&
         require('@babel/plugin-transform-flow-strip-types').default,
+      isTypeScriptEnabled && [
+        require('@babel/plugin-transform-typescript').default,
+        { isTSX: true },
```

Comment:
This breaks parsing of traditional `.ts` files that use angle bracket type casting syntax. Hopefully nobody uses that anymore when the alternative `as` syntax is valid in both `.ts` and `.tsx` files...

Reply:
Swagger codegen still use <any> IIRC

Reply:
@Kovensky @jaredpalmer I was using @babel/preset-typescript here previously, which is better imo, it checks the file extension .ts or .tsx to enable isTSX or not. But #5182 changed the flow preset to plugin so I did as well. Should I just change again to the preset or is there a way to make the plugin make this file extension check as well?

Reply:
It's a restriction of babel's config system; only presets are able to look at the input file extension.

Better to change it back to the plugin and add a comment explaining that it has to be the preset because of the differing behavior depending on extension.

Reply:
> breaks parsing of traditional .ts files that use angle bracket type casting syntax. 

Is that the only downside? Because if so, it's a deprecated language feature and easily fixable by the user

Reply:
Please make this auto-toggle like the preset: https://github.com/babel/babel/blob/850bc1d3dd868705331f963569583ce325921d5b/packages/babel-preset-typescript/src/index.js#L26-L39

Reply:
That's fine. We were fixing a very specific bug when switching from the flow preset to the plugin, so it's probably fine with TypeScript.